File: kinetiaServer/src/main/java/com/proyectoi/kinetia/dto/Advertisement.java
package com.proyectoi.kinetia.dto;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.time.Instant;


public class Advertisement {

    @JsonProperty("id")
    private Long id;
    @JsonProperty("userId")
    private Long userId;
@JsonProperty("title")
    private String title;
@JsonProperty("description")
    private String description;
@JsonProperty("location")
    private String location;


    private Instant creationDate;




    @Override
    public String toString() {
        return "AdvertisementModel{" +
                "id=" + id +
                ", user=" + userId +
                ", title='" + title + '\'' +
                ", description='" + description + '\'' +
                ", location='" + location + '\'' +
                ", creationDate=" + creationDate +
                '}';
    }
}

File: kinetiaApp/app/src/main/java/com/dam2/proyectocliente/models/Advertisement.kt
package com.dam2.proyectocliente.models

import com.dam2.proyectocliente.utils.InstantSerializer
import kotlinx.serialization.Serializable
import java.time.Instant

@Serializable
data class Advertisement(
    val id: Long? = null,
    val userId: Long,
    val userName: String,
    val userPhoto: String,
    var title: String,
    var description: String,
    @Serializable(with = InstantSerializer::class)
    val creationDate: Instant = Instant.now(),
    var location: String
) {

}